cheetah/accelerator/dipole.py

```diff
@@ -1,45 +1,62 @@
-from typing import Optional, Union
+from typing import Literal, Optional, Union
 
 import matplotlib.pyplot as plt
 import numpy as np
 import torch
 from matplotlib.patches import Rectangle
+from scipy.constants import physical_constants
 from torch import Size, nn
 
+from cheetah.particles import Beam, ParticleBeam
 from cheetah.track_methods import base_rmatrix, rotation_matrix
-from cheetah.utils import UniqueNameGenerator
+from cheetah.utils import UniqueNameGenerator, bmadx
 
 from .element import Element
 
 generate_unique_name = UniqueNameGenerator(prefix="unnamed_element")
 
+electron_mass_eV = torch.tensor(
+    physical_constants["electron mass energy equivalent in MeV"][0] * 1e6,
+    dtype=torch.float64,
+)
+
 
 class Dipole(Element):
     """
     Dipole magnet (by default a sector bending magnet).
 
     :param length: Length in meters.
+    :param p0c: Reference momentum at dipole in eV/c.
     :param angle: Deflection angle in rad.
     :param e1: The angle of inclination of the entrance face [rad].
     :param e2: The angle of inclination of the exit face [rad].
     :param tilt: Tilt of the magnet in x-y plane [rad].
     :param fringe_integral: Fringe field integral (of the enterance face).
-    :param fringe_integral_exit: (only set if different from `fint`) Fringe field
-        integral of the exit face.
-    :param gap: The magnet gap [m], NOTE in MAD and ELEGANT: HGAP = gap/2
+    :param fringe_integral_exit: (only set if different from `fint`)
+        Fringe field integral of the exit face.
+    :param gap: The magnet gap [m], NOTE in MAD and ELEGANT: HGAP = gap/2.
+    :param gap_exit: The magnet gap at the entrance [m],
+        NOTE in MAD and ELEGANT: HGAP = gap/2.
     :param name: Unique identifier of the element.
     """
 
     def __init__(
         self,
         length: Union[torch.Tensor, nn.Parameter],
+        p0c: Optional[Union[torch.Tensor, nn.Parameter]] = None,
```

Comment:
I think the dipole should also get the reference momentum from particles with `ParticleBeam.p0c`  rather than defining it as its own parameter. Is there a specific use case that you want to have dipole ref momentum other than the particle one?
https://github.com/desy-ml/cheetah/blob/c909024359dee8d2479d4ca4534d743ec974c1ab/cheetah/particles/beam.py#L299

Reply:
I think you're right. I don't see any useful cases in which the reference momentum of particles and elements should be different. I will make use of `ParticleBeam.p0c`